pkg/client/unversioned/fake/fake.go

```diff
@@ -64,9 +64,12 @@ func (c *RESTClient) Delete() *unversioned.Request {
 }
 
 func (c *RESTClient) Do(req *http.Request) (*http.Response, error) {
+	if c.Err != nil {
```

Comment:
It ensures that when c.Err is set, we do not execute a server call that is then ignored